providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -174,6 +192,18 @@ static int ml_kem_match(const void *vkey1, const void *vkey2, int selection)
     return ossl_ml_kem_pubkey_cmp(key1, key2);
 }
 
+static int ml_kem_validate(const void *vkey, int selection, int check_type)
+{
+    const ML_KEM_KEY *key = vkey;
+
+    if (!ml_kem_has(key, selection))
+        return 0;
+
```

Comment:
I assume there is no need to tests for public and private components here? They wont be accepted as input if they are the wrong size.

Reply:
Correct, what's more, the serialised forms are not even stored. So there's not anything that could be tested for length.